joint_trajectory_controller/include/joint_trajectory_controller/hold_trajectory_builder.h

```diff
@@ -0,0 +1,110 @@
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//////////////////////////////////////////////////////////////////////////////
+#ifndef HOLD_TRAJECTORY_BUILDER_H
+#define HOLD_TRAJECTORY_BUILDER_H
+
+#include <vector>
+
+#include <joint_trajectory_controller/trajectory_builder.h>
+
+namespace joint_trajectory_controller
+{
+
+/**
+ * @brief Builder creating a trajectory "simply" holding (without motion)
+ * the specified position.
+ */
+template<class SegmentImpl, class HardwareInterface>
+class HoldTrajectoryBuilder : public TrajectoryBuilder<SegmentImpl>
+{
+private:
+  using Segment               = JointTrajectorySegment<SegmentImpl>;
+  using TrajectoryPerJoint    = std::vector<Segment>;
+  using Trajectory            = std::vector<TrajectoryPerJoint>;
+
+  using RealtimeGoalHandle    = realtime_tools::RealtimeServerGoalHandle<control_msgs::FollowJointTrajectoryAction>;
+  using RealtimeGoalHandlePtr = boost::shared_ptr<RealtimeGoalHandle>;
+
+  using JointHandle = typename HardwareInterface::ResourceHandleType;
+
+public:
+  HoldTrajectoryBuilder(const unsigned int& number_of_joints,
+                        const std::vector<JointHandle>&  joints);
+
+public:
+  /**
+   * @brief See base class.
+   */
+  bool buildTrajectory(Trajectory* hold_traj) override;
+
+private:
+  const unsigned int number_of_joints_;
+  const std::vector<JointHandle>&  joints_;
+
+private: //Pre-allocated memory for real time usage of build function
+  typename Segment::State hold_start_state_ {typename Segment::State(1)};
+  typename Segment::State hold_end_state_   {typename Segment::State(1)};
+
+};
+
+template<class SegmentImpl, class HardwareInterface>
+HoldTrajectoryBuilder<SegmentImpl, HardwareInterface>::HoldTrajectoryBuilder(const unsigned int& number_of_joints,
+                                                                             const std::vector<JointHandle>& joints)
+  : number_of_joints_(number_of_joints)
+  , joints_(joints)
+{
+
+}
+
+template<class SegmentImpl, class HardwareInterface>
+bool HoldTrajectoryBuilder<SegmentImpl, HardwareInterface>::buildTrajectory(Trajectory* hold_traj)
+{
+  if(!TrajectoryBuilder<SegmentImpl>::getStartTime())
+  {
+    return false;
+  }
+
+  const typename Segment::Time start_time {TrajectoryBuilder<SegmentImpl>::getStartTime().value()};
+  RealtimeGoalHandlePtr goal_handle {TrajectoryBuilder<SegmentImpl>::createGoalHandlePtr()};
+  for (unsigned int joint_index = 0; joint_index < number_of_joints_; ++joint_index)
+  {
+    hold_start_state_.position[0]     =  joints_[joint_index].getPosition();
+    hold_start_state_.velocity[0]     =  0.0;
+    hold_start_state_.acceleration[0] =  0.0;
+
+    Segment& segment {(*hold_traj)[joint_index].front()};
```

Comment:
I think the checks were left out intentionally, because you guarantee that by construction via `createHoldTrajectory()` in the controller.
However I would not mind adding the checks. What do you think?
Also: How about moving `createHoldTrajectory()` into `hold_trajectory_builder.h`?

Reply:
>However I would not mind adding the checks. What do you think?

Please, add the checks.

> Also: How about moving createHoldTrajectory() into hold_trajectory_builder.h?

For me the `createHoldTrajectory()` function is like a factory function. If you move the function to the `hold_trajectory_builder.h`, please separate this function from the actual builder. You could add it e.g. as a stand-alone static function.

Reply:
Done.